gems/pending/spec/appliance_console/database_replication_spec.rb

```diff
@@ -0,0 +1,167 @@
+    allow(subject).to receive(:ask_for_ip_or_hostname)
+    allow(subject).to receive(:ask_for_password_or_none)
+    allow(subject).to receive(:ask_for_password)
+  end
+
+  context "#ask_for_unique_cluster_node_number" do
+    it "should ask for a unique number" do
+      expect(subject).to receive(:ask_for_integer).with(/uniquely identifying this node/i).and_return(1)
+
+      subject.ask_for_unique_cluster_node_number
+    end
+  end
+
+  context "#ask_for_database_credentials" do
+    before do
+      subject.node_number       = 'defaultnodenumber'
+      subject.database_name     = 'defaultdatabasename'
+      subject.database_user     = 'defaultuser'
+      subject.primary_host      = 'defaultprimary'
+    end
+
+    it "should default prompts (no default password)" do
+      subject.database_password = nil
+
+      expect(subject).to receive(:just_ask).with(/ user/i, 'defaultuser').and_return("newuser")
+      expect(subject).to receive(:just_ask).with(/ name/i, 'defaultdatabasename').and_return("newuser")
+      expect(subject)
+        .to receive(:ask_for_ip_or_hostname).with(/primary.*hostname/i, 'defaultprimary').and_return("newprimary")
+      expect(subject).to receive(:ask_for_password_or_none).with(/password/i, nil).and_return("x")
+      expect(subject).to receive(:ask_for_password).with(/password/i).and_return("x")
+
+      subject.ask_for_database_credentials
+    end
+
+    it "should default password prompt with stars (choosing default doesnt confirm password)" do
+      subject.database_password = "********"
+
+      expect(subject).to receive(:just_ask).with(/ username/i, 'defaultuser').and_return("newuser")
+      expect(subject).to receive(:just_ask).with(/ name/i, 'defaultdatabasename').and_return("newuser")
+      expect(subject)
+        .to receive(:ask_for_ip_or_hostname).with(/primary.*hostname/i, 'defaultprimary').and_return("newprimary")
+      expect(subject).to receive(:ask_for_password_or_none).with(/password/i, "********").and_return("********")
+
+      subject.ask_for_database_credentials
+    end
+
+    it "should ask for user/password (with confirm)" do
+      subject.database_password = "defaultpass"
+
+      expect(subject).to receive(:just_ask).with(/ username/i, 'defaultuser').and_return("newuser")
+      expect(subject).to receive(:just_ask).with(/ name/i, 'defaultdatabasename').and_return("newuser")
+      expect(subject)
+        .to receive(:ask_for_ip_or_hostname).with(/primary.*hostname/i, 'defaultprimary').and_return("newprimary")
+      expect(subject).to receive(:ask_for_password_or_none).with(/password/i, anything).and_return("the password")
+      expect(subject).to receive(:ask_for_password).with(/password/i).and_return("the password")
+
+      subject.ask_for_database_credentials
+    end
+  end
+
+  context "#confirm" do
+    it "should ask for standby host when requested" do
+      subject.standby_host = 'defaultstandby'
+
+      expect(subject).to receive(:clear_screen)
```

Comment:
Why are we making these expectations in a test that is only dealing with whether we display the `Standby Host:` message?

I feel like that is the only necessary expectation here. The others make this unnecessarily brittle.